doc/dev_guide/c_style_guide_new.rst

```diff
@@ -0,0 +1,1046 @@
+-------------------------------------------------------------------------------
+                                C Style Guide
+-------------------------------------------------------------------------------
+
+We use Git for revision control. The latest development is happening in the
+default branch (currently ``master``). Our git repository is hosted on GitHub,
+and can be checked out with ``git clone git://github.com/tarantool/tarantool.git``
+(anonymous read-only access).
+
+If you have any questions about Tarantool internals, please post them on the
+`developer discussion list <https://groups.google.com/forum/#!forum/tarantool>`_
+or on `StackOverflow <https://stackoverflow.com/questions/tagged/tarantool>`_.
+Additionally, some engineers are always present on #tarantool channel on
+irc.freenode.net.
+
+**General guidelines**
+
+The project's coding style is based on the `Linux kernel coding style
+<https://www.kernel.org/doc/html/v4.10/process/coding-style.html>`_.
+
+However, we have some additional guidelines, either unique to Tarantool or
+deviating from the Kernel guidelines. Below we cite the Linux kernel
+coding style noting Tarantool's style features.
+
+We don't cite chapters 10 "Kconfig configuration files", 11 "Data structures",
+13 "Printing kernel messages", and 17 "Don't re-invent the kernel macros" since
+they are specific to Linux kernel programming environment.
+
+================================================================================
+                           Linux kernel coding style
+================================================================================
+
+This is a short document describing the preferred coding style for the
+linux kernel. Coding style is very personal, and I won't **force** my
+views on anybody, but this is what goes for anything that I have to be
+able to maintain, and I'd prefer it for most other things too.  Please
+at least consider the points made here.
```

Comment:
This paragraph could make a user think he has a choice not to use this guide, which is not true. We exactly **force** our style.

I guess this is a direct cite from the kernel guide, which adds +1 point to the option of rewriting the guide entirely.